pandas/core/shared_docs.py

```diff
@@ -570,8 +570,11 @@
         .. deprecated:: 2.1.0
     regex : bool or same types as `to_replace`, default False
         Whether to interpret `to_replace` and/or `value` as regular
-        expressions. If this is ``True`` then `to_replace` *must* be a
-        string. Alternatively, this could be a regular expression or a
+        expressions. If this is ``True``, `value` is not ``None`` and `to_replace` is a string,
```

Comment:
Can you describe this in the `Example`s section below instead?

Reply:
Sure!

Reply:
@mroeschke please let me know if i can do any improvements/corrections!

Reply:
Please reference the code checks and pre-commit CI jobs for some formatting issues that may exist here